src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -199,6 +199,19 @@ private EmissionsForecast ProcessAndValidateForecast(EmissionsForecast forecast,
         return emissionsData.MinBy(x => x.Rating);
     }
 
+    private IEnumerable<EmissionsData> GetOptimalEmissions(IEnumerable<EmissionsData> emissionsData)
+    {
+        if (!emissionsData.Any())
+        {
+            return Array.Empty<EmissionsData>();
+        }
+
+        var bestResult = emissionsData.MinBy(x => x.Rating);
+        return (bestResult != null) ?
```

Comment:
NIT: If the ternary doesn't fit on a single line, prefer a standard conditional

```suggestion
        var results = Array.Empty<EmissionsData>();
        if (bestResult != null)
        {
            results = emissionsData.Where(x => x.Rating == bestResult.Rating);
        }
        return results;
```